src/iccpd/src/mlacp_sync_update.c

```diff
@@ -0,0 +1,659 @@
+            "    Peer port %s is removed from port-channel member.",portconf->port_name);*/
+
+        return 0;
+    }
+
+    if (pif == NULL && portconf->flags & 0x01)
+    {
+        pif = peer_if_create(csm, ntohs(portconf->agg_id), IF_T_PORT_CHANNEL);
+        if (pif == NULL)
+            return -1;
+
+        new_create = 1;
+    }
+
+    pif->po_id = ntohs(portconf->agg_id);
+    memcpy(pif->name, portconf->agg_name, portconf->agg_name_len);
+    memcpy(pif->mac_addr, portconf->mac_addr, ETHER_ADDR_LEN);
+
+    po_active = (pif->state == PORT_STATE_UP);
+    update_stp_peer_link(csm, pif, po_active, new_create);
+    update_peerlink_isolate_from_pif(csm, pif, po_active, new_create);
+    pif->po_active = po_active;
+
+    return 0;
+}
+
+/*****************************************
+* Agg Port-State Update
+*
+* ***************************************/
+int mlacp_fsm_update_Aggport_state(struct CSM* csm, mLACPAggPortStateTLV* tlv)
+{
+    struct PeerInterface* peer_if = NULL;
+    uint8_t po_active;
+
+    if (csm == NULL || tlv == NULL)
+        return -255;
+    ICCPD_LOG_DEBUG(__FUNCTION__, "  po id %d  state %d  ", ntohs(tlv->agg_id), tlv->agg_state);
+
+    po_active = (tlv->agg_state == PORT_STATE_UP);
+
+    LIST_FOREACH(peer_if, &(MLACP(csm).pif_list), mlacp_next)
+    {
+        if (peer_if->type != IF_T_PORT_CHANNEL)
+            continue;
+
+        if (peer_if->po_id != ntohs(tlv->agg_id))
+            continue;
+
+        peer_if->state = tlv->agg_state;
+
+        update_stp_peer_link(csm, peer_if, po_active, 0);
+        update_peerlink_isolate_from_pif(csm, peer_if, po_active, 0);
+
+        peer_if->po_active = po_active;
+        ICCPD_LOG_DEBUG(__FUNCTION__, "Update  Msg for %s  state %s", peer_if->name, tlv->agg_state ? "down" : "up");
+
+        break;
+    }
+
+    return 0;
+}
+
+/*****************************************
+* Recv from peer, MAC-Info Update
+* ***************************************/
+int mlacp_fsm_update_mac_info_from_peer(struct CSM* csm, struct mLACPMACInfoTLV* tlv)
+{
+    struct Msg* msg = NULL;
+    struct MACMsg *mac_msg = NULL, mac_data;
+    struct LocalInterface* local_if = NULL;
+    uint8_t from_mclag_intf = 0;/*0: orphan port, 1: MCLAG port*/
+
+    if (!csm || !tlv)
+        return -255;
+
+#if 1
+    ICCPD_LOG_INFO(__FUNCTION__,
+                   "Received MAC Info, itf=[%s] vid[%d] MAC[%s]  type %d ",
+                   tlv->ifname, ntohs(tlv->vid), tlv->mac_str, tlv->type);
+#endif
+
+    /*Find the interface in MCLAG interface list*/
+    LIST_FOREACH(local_if, &(MLACP(csm).lif_list), mlacp_next)
+    {
+        if (local_if->type == IF_T_PORT_CHANNEL && strcmp(local_if->name, tlv->ifname) == 0)
+        {
+            from_mclag_intf = 1;
+            break;
+        }
+    }
+
+    if (!local_if)
+    {
+        if (!(local_if = local_if_find_by_name(tlv->ifname)))
+        {
+            /*If the interface is not exist, return*/
+            return -255;
+        }
+    }
```

Comment:
does the Peer node interface is part of local If .? I assume not. Then this check would fail for orphan port mac learn if the incoming interface is not present here. 
I see someone posted the same comment , please clarify.?

Reply:
@Praveen-Brcm, you are right. For orphan port, name check is not needed.